### ISSUE TITLE
Organizations want to be able to specify the format of the names of files they receive

### ISSUE THREAD

**Issue by MauriceReeves-usds** [CDCgov/prime-data-hub#462]
LA has let us know that they have a specific file name format that they would like us to use when we upload. The documentation is attached for review. We should make it possible for organizations to define this.[https://app.zenhub.com/files/304423150/d8dcd7a6-fe9f-4a40-b67a-cbf1222461ec/download](https://app.zenhub.com/files/304423150/d8dcd7a6-fe9f-4a40-b67a-cbf1222461ec/download)

**Comment by jimduff-usds** [comment]
Yes, this can be done.   I'd implement this similar to the `jurisdictionalFilters` feature, anticipating lots of one-of requests that can't easily be put into a configuration language.

Just as an example, we had this request from Harris county:  "Remove the symbol “–“ from file name and sheet name as SAS cannot recognize or open files with those characters. Recommend using “_” or “.” to demarcate spaces in titles."

@mauricereeves , my main question is around what they mean by 'prod' and 'test'.  Is their test system for junk/fake/synthetic data?  (That would imply we wire our Test system to theirs). Or, is it more like AZ PHD, where Test is real live data, but just not fully validated yet. (That would imply we create two `receivers` for LA, both in our Prod, but one labelled la-test and one la-prod -- this is what we do for AZ)

**Comment by jimduff-usds** [comment]
A few more questions:
- there is a 'file_created_date` that goes into MSH-7 - we would need that to be used and filled in, then we can just extract it, and use it for the timestamp.   Is that field used in the LA schema?
- We'd need to know what they mean by 'sending org' and 'sending facility'.   I'm guessing neither of these is "Simple Report".   Ideally, these would be data in the schema, coming from SR, so we can just extract it.   
- The sending org can be organization_name, currently in the pdi schema, but we can factor it up a level.
- The sending facility can be put in `ordering_facility_name`
- Since the values they want in the filename may vary per result, **I'm hoping LA would be ok with each hl7 being a separate file,** rather than bundled.  Otherwise, in order to batch them, I think we'll have to create different "receiver"s in the organizations.yml for every sending org/facility.

**Comment by MauriceReeves-usds** [comment]
@jimduff-usds 

1. I will actually need to confirm with LA what they want to see in the file. Based on the fact that we're sending them test communications, my sense is we should send them D, but once we go live it flips to P. But I will ask them to confirm that logic. They are the only state so far to request this field be filled in, so many states might not even care. I know the CDC said most states just test with live data anyway, so if we wanted to stand them up in the test system for debugging reasons as we did with AZ, I'm okay with that.
2. For now I'm setting `MSH-7` to be the date the file is serialized down into HL7, but if Simple Report wants to send us the `file_created_date` we can use that instead.
3. Sending organization is actually Hub in our case, but they want `sending_facility` to be the lab that filed the report. I was overriding `sending_facility` to also be Hub, but I've removed that and now the `sending_facility` field in the MSH segment is showing the facility information, so this is fixed.

I did some work on #453 last night that addresses most of the MSH segment issues.

**Comment by jlusds** [comment]
Was this completed?

**Comment by RickHawesUSDS** [comment]
Here's the CA DPH filename standard. 

Naming Standard: LabName_DateSubmittedToCDPH_FileCount.csv

- LabName CAN be abbreviated, but must NOT include any spaces or separators such as dashes or underscores

- DateSubmittedToCDPH is formatted YYYYMMDD and is the date the file is actually sent to CDPH

- FileCount is single digit (1, 2, 3) to allow for multiple CSV files on a given day, and must always be included

Example: AcmeLab_20200108_1.csv (AcmeLab_20200108_2.csv if submitting two files the same day)

**Comment by RickHawesUSDS** [comment]
@mauricereeves

**Comment by MauriceReeves-usds** [comment]
@jimduff-usds based on the above req's for the file name for CA, it feels like the lineage system might be the easiest way to get the file count by organization/receiver. Thoughts?
